src/Form/Mixin/Value.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Kirby\Form\Mixin;
+
+/**
+ * @package   Kirby Form
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://opensource.org/licenses/MIT
+ */
+trait Value
+{
+	/**
+	 * @deprecated 3.5.0 Use `::toStoredValue()` instead
+	 * @todo remove when the general field class setup has been refactored
+	 */
+	public function data(bool $default = false): mixed
+	{
+		return $this->toStoredValue($default);
+	}
+
+	/**
+	 * Returns the default value of the field
+	 */
+	public function default(): mixed
+	{
+		if (is_string($this->default) === false) {
+			return $this->default;
+		}
+
+		return $this->model->toString($this->default);
+	}
+
+	/**
+	 * Checks if the field is empty
+	 */
+	public function isEmpty(): bool
+	{
+		return $this->isEmptyValue($this->toFormValue());
+	}
+
+	/**
+	 * Checks if the given value is considered empty
+	 */
+	public function isEmptyValue(mixed $value = null): bool
+	{
+		return in_array($value, [null, '', []], true);
```

Comment:
`Question:` Wondering if this actually would need to mirror `Content\Field::isEmpty()`?

Reply:
Hm, that's a very good question. I'm not sure if we break something unexpected when we start trimming strings before checking here.

Reply:
Maybe not trimming, but check for `'[]'` as well?

Reply:
But we can also leave it as is for now, so this is not the only thing left blocking merging.

Reply:
We just recently added this because of https://github.com/getkirby/kirby/issues/6637 - that's how the Symphony YAML parser stores empty arrays.

Reply:
Not sure I would go there, transforming each field now in storage first would be a big change. But I'll mark this as resolved for this PR then.